tests/routes/movies.test.ts

```diff
@@ -0,0 +1,198 @@
+
+const movie: Movie = {
+  id: 'movieId',
+  title: 'Movie',
+  description: 'Movie description',
+  releaseDate: new Date('2021-05-08'),
+  genre: ['Action', 'Fantasy'],
+} as Movie;
+
+describe('Movies route', () => {
+  beforeAll((done) => {
+    server.close();
+    server.listen(PORT, () => {
+      console.log(`Test server is listening on port ${PORT}`);
+      done();
+    });
+  });
+
+  describe('getAllMovies', () => {
+    it('should return a list of movies', async () => {
+      const fakeMovies: Movie[] = [
+        {
+          title: 'Movie_1',
+          description: 'Movie_1 description',
+        },
+        {
+          title: 'Movie_2',
+          description: 'Movie_2 description',
+        },
+      ] as Movie[];
+      movieModel.find = jest.fn().mockResolvedValue(fakeMovies);
+      const response = await request.get('/movies');
+      expect(response.status).toBe(200);
+      expect(response.body).toEqual(fakeMovies);
+    });
+
+    it('should return a 500 error', async () => {
+      movieModel.find = jest.fn().mockRejectedValue(new Error('Internal Server Error'));
+      const response = await request.get('/movies');
+      expect(response.status).toBe(500);
+      expect(response.body).toEqual({ error: 'Internal Server Error' });
+    });
+  });
+
+  describe('createMovie', () => {
+    it('should create a new movie with valid fields', async () => {
+      const validMovieData: Movie = {
+        title: 'New movie',
+        description: 'New movie description',
+        releaseDate: new Date('2021-05-08'),
+        genre: ['Action', 'Fantasy'],
+      };
+      const expectedMovieData = { ...validMovieData, releaseDate: validMovieData.releaseDate.toISOString() };
+      const response = await request.post('/movies').send(expectedMovieData);
+      expect(response.status).toBe(201);
+      expect(response.body).toEqual(expectedMovieData);
+    });
+
+    it('should handle invalid fields and return a 400 error', async () => {
+      const invalidMovieData: Movie = {
+        title: '',
+        description: 'New movie description',
+        releaseDate: new Date('2021-05-08'),
+        genre: ['Action', 'Fantasy'],
+      };
+      const response = await request.post('/movies').send(invalidMovieData);
+      expect(response.status).toBe(400);
+      expect(response.body).toEqual({ error: 'Title field is required' });
+    });
+
+    it('should return a 422 error for invalid date format', async () => {
+      const invalidMovieData: Movie = {
+        title: 'New Movie',
+        description: 'New Movie Description',
+        releaseDate: new Date('2023-01-xx'),
+        genre: ['Comedy', 'Fantasy'],
+      };
+      const expectedMovieData = { ...invalidMovieData, releaseDate: invalidMovieData.releaseDate.toDateString() };
+      const response = await request.post('/movies').send(expectedMovieData);
+      expect(response.status).toBe(422);
+      expect(response.body).toEqual({ error: 'Invalid release date' });
```

Comment:
Consider to move error message to some constant, enum, class ....
If you would like to change error message, you will need to change everywhere in each place where you use it